src/guid.cpp

```diff
@@ -0,0 +1,48 @@
+// Luanti
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2024 SFENCE
+
+#include "guid.h"
+#include <cstring>
+#include <sstream>
+#include <string_view>
+
+#include "exceptions.h"
+#include "util/base64.h"
+#include "log.h"
+
+std::string MyGUID::base64() const
+{
+	return base64_encode(std::string_view(&bytes[0], bytes.size()));
+}
+
+void MyGUID::serialize(std::ostringstream &os) const
+{
+	os.write(&bytes[0], bytes.size());
+}
+
+void MyGUID::deSerialize(std::istream &is)
+{
+	is.read(&bytes[0], bytes.size());
+}
+
+GUIDGenerator::GUIDGenerator() :
+	m_uniform(0, UINT64_MAX)
+{
+	if (m_rand.entropy() <= 0.01)
+		warningstream <<
+				"The system's provided random generator reports low entropy."
+				"GUID generator can be affected. Suggest a system upgrade."
+				<< std::endl;
+}
+
+MyGUID GUIDGenerator::next()
+{
+	u64 rand1 = m_uniform(m_rand);
+	u64 rand2 = m_uniform(m_rand);
```

Comment:
are we sure we should be using the random device directly with `std::uniform_int_distribution`?
see the claim in the example code here: https://en.cppreference.com/w/cpp/numeric/random/random_device#Example

Reply:
i was originally skeptical as well. i think the reason @sfence wants to use the random device directly is that merely using it to seed a pseudorandom could lose us too much entropy. sfence did a quick benchmark (with what i assume are enough iterations to exhaust the entropy pool) and it does not seem that this is likely to be a performance bottleneck: https://github.com/luanti-org/luanti/pull/14135#discussion_r2029908065

Reply:
Whether or not reading directly from the random device works well is OS-dependent so I think this is not an assumption we should make.